### DISCUSSION TITLE
ModelCheckpoint creating unexpected subfolders

### DISCUSSION THREAD

**Discussion by jiwidi** [Lightning-AI/pytorch-lightning#8300] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi! I have found a weird behavior when using ModelCheckpoint, if I have a metric that I want to save in my filename and it has a "/" on it it will create nested directories. For example

```py
checkpoint_callback = ModelCheckpoint(
        monitor='val/acc',
        dirpath=checkpoints_dir,
        filename='checkpoint_{epoch:02d}-{val/acc}',
        save_top_k=-1,
     )
```
This one will create one extra folder per checkpoint:
```bash
checkpoints/base_lstm/checkpoint_epoch=00-val/acc=0.04-v1.ckp
checkpoints/base_lstm/checkpoint_epoch=00-val/acc=0.05-v1.ckp
```
Is there any way to make the modelcheckpoint callback store the "val/acc" value while not using the string "val/acc" to reference it? Something like:
checkpoints/base_lstm/checkpoint_epoch=00-valAcc=0.04-v1.ckp
I think is quite standard to use "/" on tensorboard to be able to use the inbuilt tabs to better group metrics.

**Comment by Borda** [comment, marked as answer]
hi, can you provide also your model sample, in particular, the metrics section
I guess that the problem is with `/` as it is interpreted as a normal folder path, as you can see that `'val/acc'` is not replaced by a number either... :rabbit:

**Reply by jiwidi** [reply]
Hi! This actually got solved in the slack. Turns out there is a parameter for it but didnt saw it since it was missing documentation about it, I opened a PR to solve that https://github.com/PyTorchLightning/pytorch-lightning/pull/8310
